seleniumwire/proxy/modifier.py

```diff
@@ -99,8 +115,17 @@ def modify(self, request):
         self._rewrite_url(request)
 
     def _modify_headers(self, request):
+        # If self._headers is tuple or list, need to use the pattern matching
+        if is_list_alike(self._headers):
+            headers = self._matched_headers(self._headers, request.path)
+        else:
+            headers = self._headers
+
+        if not headers:
+            return
+
```

Comment:
I think we should move this section so that it inside `with self._lock:`

Reply:
You're totally right. I have updated it